src/types/error.rs

```diff
@@ -1,70 +1,140 @@
-use types::{Value, Object, Handle, Managed, build};
 use types::internal::ValueInternal;
 use types::utf8::Utf8;
+use types::{build, Handle, Managed, Object, Value};
 
 /// A JS `Error` object.
 #[repr(C)]
 #[derive(Clone, Copy)]
 pub struct JsError(raw::Local);
 
 impl Managed for JsError {
-    fn to_raw(self) -> raw::Local { self.0 }
+    fn to_raw(self) -> raw::Local {
+        self.0
+    }
 
-    fn from_raw(h: raw::Local) -> Self { JsError(h) }
+    fn from_raw(h: raw::Local) -> Self {
+        JsError(h)
+    }
 }
 
 impl ValueInternal for JsError {
-    fn name() -> String { "Error".to_string() }
+    fn name() -> String {
+        "Error".to_string()
+    }
 
     fn is_typeof<Other: Value>(other: Other) -> bool {
         unsafe { neon_runtime::tag::is_error(other.to_raw()) }
     }
 }
 
-impl Value for JsError { }
+impl Value for JsError {}
 
-impl Object for JsError { }
+impl Object for JsError {}
 
 impl JsError {
     /// Creates a direct instance of the [`Error`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Error) class.
-    pub fn error<'a, C: Context<'a>, S: AsRef<str>>(cx: &mut C, msg: S) -> NeonResult<Handle<'a, JsError>> {
+    pub fn error<'a, C: Context<'a>, S: AsRef<str>>(
+        cx: &mut C,
+        msg: S,
+    ) -> NeonResult<Handle<'a, JsError>> {
+        #[cfg(feature = "legacy-runtime")]
         let msg = cx.string(msg.as_ref());
+
+        #[cfg(feature = "napi-runtime")]
+        let (ptr, len) = if let Some(small) = Utf8::from(msg.as_ref()).into_small() {
+            small.lower()
+        } else {
+            return Err(Throw);
+        };
         build(|out| unsafe {
+            #[cfg(feature = "napi-runtime")]
+            let mut local: raw::Local = std::mem::zeroed();
+            #[cfg(feature = "napi-runtime")]
+            neon_runtime::string::new(&mut local, cx.env().to_raw(), ptr, len);
+            #[cfg(feature = "napi-runtime")]
+            neon_runtime::error::new_error(out, cx.env().to_raw(), std::ptr::null_mut(), local);
+            #[cfg(feature = "legacy-runtime")]
             neon_runtime::error::new_error(out, msg.to_raw());
             true
         })
     }
 
     /// Creates an instance of the [`TypeError`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/TypeError) class.
-    pub fn type_error<'a, C: Context<'a>, S: AsRef<str>>(cx: &mut C, msg: S) -> NeonResult<Handle<'a, JsError>> {
+    pub fn type_error<'a, C: Context<'a>, S: AsRef<str>>(
+        cx: &mut C,
+        msg: S,
+    ) -> NeonResult<Handle<'a, JsError>> {
+        #[cfg(feature = "legacy-runtime")]
         let msg = cx.string(msg.as_ref());
+
+        #[cfg(feature = "napi-runtime")]
+        let (ptr, len) = if let Some(small) = Utf8::from(msg.as_ref()).into_small() {
+            small.lower()
+        } else {
+            return Err(Throw);
+        };
         build(|out| unsafe {
+            #[cfg(feature = "napi-runtime")]
```

Comment:
Again, collapse the three identical `#[cfg(...)]` branches into one by grouping the three statements into a block.